src/main.rs

```diff
@@ -1,11 +1,17 @@
-mod fen;
-mod game_state;
+use game_state::GameState;
+use std::str::FromStr;
+
 mod board;
+mod game_state;
 
 fn main() {
-    
-    let fen_string = "rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1";
-    let game_state = fen::fen_to_board(fen_string);
+    let game_state =
+        GameState::from_str("rnbqkbnr/pppppppp/8/8/8/8/PPPPPPPP/RNBQKBNR w KQkq - 0 1");
+    if let Err(e) = game_state {
+        println!("Error: {}", e);
+        return;
+    }
+    let game_state = game_state.unwrap();
```

Comment:
can't we use else here?
is this the best way for to get the Ok? unwrap is dirty

Reply:
Even if we use else here we still need to unwrap() so it would look like this -->
 ` if let Err(e) = game_state {
        println!("Error: {}", e);
        return;
    } else {
        game_state.unwrap().board.print_board();
    }`

Else is not really needed however because we "return" inside the previous if. Some linters will even mark this as unused.